Dockerfile

```diff
@@ -1,60 +1,57 @@
-ARG alpine_version=3.12
-ARG go_version=1.14
+ARG debian=buster
+ARG go_version
 ARG grpc_version
 ARG grpc_java_version
-ARG proto_version
 
-FROM golang:$go_version-alpine$alpine_version AS build
+FROM golang:$go_version-$debian AS build
 
 # TIL docker arg variables need to be redefined in each build stage
 ARG grpc_version
 ARG grpc_java_version
 ARG grpc_web_version
-ARG proto_version
 
-RUN set -ex && apk --update --no-cache add \
-    bash \
-    make \
+RUN set -ex && apt-get update && apt-get install -y --no-install-recommends \
+    build-essential \
+    pkg-config \
     cmake \
-    autoconf \
-    automake \
     curl \
-    tar \
-    libtool \
-    g++ \
     git \
-    openjdk8-jre \
-    libstdc++ \
-    ca-certificates \
-    nss \
-    linux-headers \
-    unzip
+    openjdk-11-jre \
+    unzip \
+    libtool \
+    autoconf \
+    zlib1g-dev \
+    libssl-dev \
+    clang
+
+WORKDIR /tmp
 
-RUN set -ex && apk --update --no-cache add \
-    protoc~=${proto_version} \
-    protobuf~=${proto_version} \
-    protobuf-dev~=${proto_version} \
-    --repository=http://dl-cdn.alpinelinux.org/alpine/edge/main
+RUN git clone -b v$grpc_version.x --recursive -j8 --depth 1 https://github.com/grpc/grpc && \ 
+    git clone -b v$grpc_java_version.x --recursive https://github.com/grpc/grpc-java.git && \
+    git clone -b v$grpc_version.x --recursive https://github.com/grpc/grpc-go.git && \
+    git clone https://github.com/googleapis/googleapis && \
+    git clone https://github.com/googleapis/api-common-protos
```

Comment:
what about `--depth 1` for all of these?

Reply:
sounds good to me